### ISSUE TITLE
2019-12-06 M3SD - remove on parameter at a time from PARAMETERLIST to identify sensitive parameters for which proper dependency management is not occurring leading to crashes

### ISSUE THREAD

**Issue by tensfeldt** [tensfeldt/openNCA#91]
## 2019-12-06_M3SD.json
```
[1] "AUCTDN;AUCXPCTO;AUCXPCTP;AUMCINFO;AUMCINFP;AUMCLAST;AUMCXPTO;AUMCXPTP;CENDINF;CENDINFDN;CEST;CLAST;CLO;CLOW;CLP;CLPW;CMAX;CMAXDN;CMIN;CMINDN;DOFi;DOSE;DOSEC;KEL;KELC0;KELNOPT;KELR;KELRSQ;KELRSQA;KELTMHI;KELTMLO;LASTTIME;MRTIVIFO;MRTIVIFP;MRTLAST;TENDINF;THALF;THALFF;TLAST;TMAX;TMIN;VSSP;VSSPW;VSSO;VSSOW;VZO;VZOW;VZP;VZPW;FLGACCEPTPREDOSE;FLGACCEPTTMAX;FLGACCEPTKEL"

```

## i:  AUCT1_T2  removed
```
unit_conversion : Unit Class 1 (Time) time_col:  TENDINF TMAX TMIN TLAST KELTMLO KELTMHI THALF LASTTIME MRTLAST MRTIVIFO MRTIVIFP  parameters are scaled from  HR  to  HR  via scaling factor:  1 
unit_conversion : Unit Class 3 (Dose) dose_col:  DOSE  parameters are scaled from  UG  to  MG  via scaling factor:  0.001 
unit_conversion : Unit Class 4 (Volume) volume_col:  VZO VZP VSSO VSSP VSSPW  parameters are scaled from  ML  to  L  via scaling factor:  0.001 
unit_conversion : Unit Class 5 (Amount/Volume) conc_col:  CENDINF CMAX CMIN CLAST CEST KELC0  parameters are scaled from  NG/ML  to  NG/ML  via scaling factor:  1 
unit_conversion : Unit Class 6 (1/Time) kel_col:  KEL  parameters are scaled from  1/HR  to  1/HR  via scaling factor:  1 
unit_conversion : Unit Class 7 (Volume/Time) cl_col:  CLO CLP  parameters are scaled from  ML/HR  to  ML/HR  via scaling factor:  1 
Error in unit_conversion(data = data_data, map = map_data, result = computation_df,  : 
  object 'auc_unit_tmp' not found
In addition: There were 50 or more warnings (use warnings() to see the first 50)
```

## i:  CENDINF  removed
```
[1] "CENDINFDN;CEST;CLAST;CLO;CLOW;CLP;CLPW;CMAX;CMAXDN;CMIN;CMINDN;DOFi;DOSE;DOSEC;KEL;KELC0;KELNOPT;KELR;KELRSQ;KELRSQA;KELTMHI;KELTMLO;LASTTIME;MRTIVIFO;MRTIVIFP;MRTLAST;TENDINF;THALF;THALFF;TLAST;TMAX;TMIN;VSSP;VSSPW;VSSO;VSSOW;VZO;VZOW;VZP;VZPW;FLGACCEPTPREDOSE;FLGACCEPTTMAX;FLGACCEPTKEL;AUCT1_T2"
Error in value[[3L]](cond) : 
  Error in cendinf_dn(cendinf = cend_inf, dose = tmp_dose): object 'cend_inf' not found
For SDEID 12659123
In addition: There were 50 or more warnings (use warnings() to see the first 50)
```

## i:  TLAST  removed
```
[1] "TMAX;TMIN;VSSP;VSSPW;VSSO;VSSOW;VZO;VZOW;VZP;VZPW;FLGACCEPTPREDOSE;FLGACCEPTTMAX;FLGACCEPTKEL;AUCT1_T2;CENDINF"
Error in value[[3L]](cond) : 
  Error in cest(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 't_last' not found
For SDEID 12659123
In addition: There were 50 or more warnings (use warnings() to see the first 50)
```

## i:  TMAX  removed - this is an order of execution problem
```
[1] "TMIN;VSSP;VSSPW;VSSO;VSSOW;VZO;VZOW;VZP;VZPW;FLGACCEPTPREDOSE;FLGACCEPTTMAX;FLGACCEPTKEL;AUCT1_T2;CENDINF;TLAST"
Error in value[[3L]](cond) : 
  Error in `[<-.data.frame`(`*tmp*`, i, , value = c(12659123, 26700, 1, : replacement has 49 items, need 51
For SDEID 12659123
In addition: Warning messages:
1: In run_M3_SD_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'KELRSQ'
2: In run_M3_SD_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'KELNOPT'
3: In run_M3_SD_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'AUCXPCTP'
4: In run_M3_SD_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGEMESIS' is not present in the dataset
5: In run_M3_SD_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTPREDOSECRIT' cannot be computed if 'CMAX' is not part of the calculated parameters
6: In run_M3_SD_computation(data = merged_data, map = map_data, method = method,  :
 
 Show Traceback
 
 Rerun with Debug
 Error in value[[3L]](cond) : 
  Error in `[<-.data.frame`(`*tmp*`, i, , value = c(12659123, 26700, 1, : replacement has 49 items, need 51
For SDEID 12659123
```

## i:  FLGACCEPTKEL  removed - this flag doesn't seem to have clear dependency control
```
[1] "AUCT1_T2;CENDINF;TLAST;TMAX"
Error in value[[3L]](cond) : 
  Error in grep(x, parameter_list, ignore.case = TRUE, perl = TRUE): object 'kel_v' not found
For SDEID 12659123
In addition: There were 50 or more warnings (use warnings() to see the first 50)
```

## i:  TLAST  removed
```
[1] "TMAX;FLGACCEPTKEL"
Error in value[[3L]](cond) : 
  Error in cest(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 't_last' not found
For SDEID 12659123
In addition: There were 33 warnings (use warnings() to see them)
```

## i:  AUCT1_T2  removed with commit fce7a3c - issue remains
```
[1] "AUCTDN;AUCXPCTO;AUCXPCTP;AUMCINFO;AUMCINFP;AUMCLAST;AUMCXPTO;AUMCXPTP;CENDINF;CENDINFDN;CEST;CLAST;CLO;CLOW;CLP;CLPW;CMAX;CMAXDN;CMIN;CMINDN;DOFi;DOSE;DOSEC;KEL;KELC0;KELNOPT;KELR;KELRSQ;KELRSQA;KELTMHI;KELTMLO;LASTTIME;MRTIVIFO;MRTIVIFP;MRTLAST;TENDINF;THALF;THALFF;TLAST;TMAX;TMIN;VSSP;VSSPW;VSSO;VSSOW;VZO;VZOW;VZP;VZPW;FLGACCEPTPREDOSE;FLGACCEPTTMAX;FLGACCEPTKEL"
unit_conversion : Unit Class 1 (Time) time_col:  TENDINF TMAX TMIN TLAST KELTMLO KELTMHI THALF LASTTIME MRTLAST MRTIVIFO MRTIVIFP  parameters are scaled from  HR  to  HR  via scaling factor:  1 
unit_conversion : Unit Class 3 (Dose) dose_col:  DOSE  parameters are scaled from  UG  to  MG  via scaling factor:  0.001 
unit_conversion : Unit Class 4 (Volume) volume_col:  VZO VZP VSSO VSSP VSSPW  parameters are scaled from  ML  to  L  via scaling factor:  0.001 
unit_conversion : Unit Class 5 (Amount/Volume) conc_col:  CENDINF CMAX CMIN CLAST CEST KELC0  parameters are scaled from  NG/ML  to  NG/ML  via scaling factor:  1 
unit_conversion : Unit Class 6 (1/Time) kel_col:  KEL  parameters are scaled from  1/HR  to  1/HR  via scaling factor:  1 
unit_conversion : Unit Class 7 (Volume/Time) cl_col:  CLO CLP  parameters are scaled from  ML/HR  to  ML/HR  via scaling factor:  1 
Error in unit_conversion(data = data_data, map = map_data, result = computation_df,  : 
  object 'auc_unit_tmp' not found
In addition: There were 50 or more warnings (use warnings() to see the first 50)
```

**Comment by jhhughes256** [comment]
### Update

Testing in progress

- [ ] AUCT1_T2 removed "AUCTDN;AUCXPCTO;AUCXPCTP;AUMCINFO;AUMCINFP;AUMCLAST;AUMCXPTO;AUMCXPTP;CENDINF;CENDINFDN;CEST;CLAST;CLO;CLOW;CLP;CLPW;CMAX;CMAXDN;CMIN;CMINDN;DOFi;DOSE;DOSEC;KEL;KELC0;KELNOPT;KELR;KELRSQ;KELRSQA;KELTMHI;KELTMLO;LASTTIME;MRTIVIFO;MRTIVIFP;MRTLAST;TENDINF;THALF;THALFF;TLAST;TMAX;TMIN;VSSP;VSSPW;VSSO;VSSOW;VZO;VZOW;VZP;VZPW;FLGACCEPTPREDOSE;FLGACCEPTTMAX;FLGACCEPTKEL"
- [ ] CENDINF removed "CENDINFDN;CEST;CLAST;CLO;CLOW;CLP;CLPW;CMAX;CMAXDN;CMIN;CMINDN;DOFi;DOSE;DOSEC;KEL;KELC0;KELNOPT;KELR;KELRSQ;KELRSQA;KELTMHI;KELTMLO;LASTTIME;MRTIVIFO;MRTIVIFP;MRTLAST;TENDINF;THALF;THALFF;TLAST;TMAX;TMIN;VSSP;VSSPW;VSSO;VSSOW;VZO;VZOW;VZP;VZPW;FLGACCEPTPREDOSE;FLGACCEPTTMAX;FLGACCEPTKEL;AUCT1_T2"
- [ ] TLAST removed "TMAX;TMIN;VSSP;VSSPW;VSSO;VSSOW;VZO;VZOW;VZP;VZPW;FLGACCEPTPREDOSE;FLGACCEPTTMAX;FLGACCEPTKEL;AUCT1_T2;CENDINF"
- [ ] TMAX removed - this is an order of execution problem "TMIN;VSSP;VSSPW;VSSO;VSSOW;VZO;VZOW;VZP;VZPW;FLGACCEPTPREDOSE;FLGACCEPTTMAX;FLGACCEPTKEL;AUCT1_T2;CENDINF;TLAST"
- [ ] FLGACCEPTKEL removed - this flag doesn't seem to have clear dependency control "AUCT1_T2;CENDINF;TLAST;TMAX"
- [ ] TLAST removed "TMAX;FLGACCEPTKEL"
- [ ] AUCT1_T2 removed with commit fce7a3c - issue remains "AUCTDN;AUCXPCTO;AUCXPCTP;AUMCINFO;AUMCINFP;AUMCLAST;AUMCXPTO;AUMCXPTP;CENDINF;CENDINFDN;CEST;CLAST;CLO;CLOW;CLP;CLPW;CMAX;CMAXDN;CMIN;CMINDN;DOFi;DOSE;DOSEC;KEL;KELC0;KELNOPT;KELR;KELRSQ;KELRSQA;KELTMHI;KELTMLO;LASTTIME;MRTIVIFO;MRTIVIFP;MRTLAST;TENDINF;THALF;THALFF;TLAST;TMAX;TMIN;VSSP;VSSPW;VSSO;VSSOW;VZO;VZOW;VZP;VZPW;FLGACCEPTPREDOSE;FLGACCEPTTMAX;FLGACCEPTKEL"

**Comment by rudraya** [comment]
This issue was addressed in a377a56, please let me know if it still occurs. I tested again using the parameters list provided in the above comment for when:
- AUCT1_T2 removed
- CENDINF removed
- TMAX removed

**Comment by jhhughes256** [comment]
### Update

As of a377a5 none of the tests listed here result in a computation error.
